analyzere/utils.py

```diff
@@ -92,3 +94,18 @@ def vectorize(values):
     if isinstance(values, list):
         return ','.join(str(v) for v in values)
     return values
+
+
+def vectorize_range(values):
+    """
+    This function is for url encoding.
+    Takes a value or a tuple or list of tuples and returns a single result,
+    tuples are joined by "," if necessary, elements in tuple are joined by '_'
+    """
+    if isinstance(values, tuple):
+        return '_'.join(str(i) for i in values)
+    elif isinstance(values, list):
+        if all([isinstance(item, tuple) for item in values]):
+            return ','.join('_'.join(str(i) for i in v) for v in values)
+        raise InvalidProbabilityError('Items in the list must be tuples')
```

Comment:
I don't like `InvalidProbabilityError` for this. 

I presume that `vectorize_range()` was written and named this way to generalize the elements that are passed in. i.e. this function is not just for lists of probabilities. 

It would be nice for `window_var()`, etc. to catch an exception from this and, in turn, throw an `InvalidProbabilityError`.

Reply:
You could also consider raising a built-in exception: e.g. TypeError, ValueError

https://docs.python.org/3/library/exceptions.html

Reply:
@hanz-analyzere `vectorize_range()` is used in `window_metrics()`, `window_co_metrics()`, `window_var()`, so it will be needed to add try catch in all three functions, which will make a little bit duplicates. Or should i raise `TypeError` that @ashleylgeorge mentions?

Reply:
`TypeError` is fine.

Reply:
cool